File: Data Structures & Algorithms/is-palindrome/submission-1.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        i=0
        j=i
        while i < len(s)//2 and j < len(s):
            last = len(s)-j-1
            first = i
        
            if not s[last].isalnum():
                j+=1
                continue
            if not s[first].isalnum():
                i+=1
                continue
            if s[first].lower() != s[last].lower():
                return False
            i+=1
            j+=1
        return True

File: Data Structures & Algorithms/valid-palindrome-ii/submission-1.cpp
class Solution {
public:
    bool validPalindrome(string s) {
        int l = 0, r = s.length()-1; 
        bool skip = false; 
        while (l<r) {
            if (s[l] == s[r]) {
                l++;
                r--; 
            } else {
                return isPalindrome(s, l+1, r) || isPalindrome(s, l, r-1);
            }
        }
        return true; 
        
    }

private:
    bool isPalindrome(string& s, int l, int r) {

        while (l < r) {
            if (s[l] != s[r]) {
                return false;
            }
            l++;
            r--;
        }
        return true; 
    }
};